File: frontend/lib/features/delivery/models/delivery.dart
import '../../auth/models/user.dart';

class Delivery {
  final int id;
  final String status;
  final String pickupAddress;
  final String deliveryAddress;
  final DateTime? estimatedArrivalAt;
  final User? driver;
  final User? customer;

  Delivery({
    required this.id,
    required this.status,
    required this.pickupAddress,
    required this.deliveryAddress,
    this.estimatedArrivalAt,
    this.driver,
    this.customer,
  });

  factory Delivery.fromJson(Map<String, dynamic> json) {
    return Delivery(
      id: json['id'],
      status: json['status'],
      pickupAddress: json['pickup_address'],
      deliveryAddress: json['delivery_address'],
      estimatedArrivalAt: json['estimated_arrival_at'] != null 
          ? DateTime.parse(json['estimated_arrival_at']) 
          : null,
      driver: json['driver'] != null ? User.fromJson(json['driver']) : null,
      customer: json['customer'] != null ? User.fromJson(json['customer']) : null,
    );
  }
}

File: backend/app/Models/Delivery.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Delivery extends Model
{
    use HasFactory;

    protected $fillable = [
        'customer_id', 'driver_id', 'status', 'pickup_address',
        'delivery_address', 'estimated_arrival_at', 'started_at', 'delivered_at'
    ];

    protected $casts = [
        'estimated_arrival_at' => 'datetime',
        'started_at' => 'datetime',
        'delivered_at' => 'datetime',
    ];

    public function customer()
    {
        return $this->belongsTo(User::class, 'customer_id');
    }

    public function driver()
    {
        return $this->belongsTo(User::class, 'driver_id');
    }

    public function statusHistories()
    {
        return $this->hasMany(DeliveryStatusHistory::class);
    }

    public function driverLocations()
    {
        return $this->hasMany(DriverLocation::class);
    }

    public function issues()
    {
        return $this->hasMany(DeliveryIssue::class);
    }
}
